workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/SingletonScriptableObject.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace Improbable.Gdk.BuildSystem.Configuration
+{
+    public abstract class SingletonScriptableObject<TSelf> : ScriptableObject
+        where TSelf : SingletonScriptableObject<TSelf>
+    {
+        private static readonly List<TSelf> Instances =
+            new List<TSelf>();
+        
+        private static readonly HashSet<Type> LoadingInstances = new HashSet<Type>();
```

Comment:
@firtina-improbable  moved this from the loader class into here. seems to work just fine. was there a reason to have it in a separate class?

Reply:
I think previously the reason was this:

Each "non-generic implementation" of this generic class would have its own copy of `LoadingInstances` rather than "one hashset to contain them all". For example `SingletonScriptableObject<Foo>.LoadingInstances != SingletonScriptableObject<Bar>.LoadingInstances`.

If it is declared within the implementation, it'd effectively be the same thing as `static bool IsLoading` - since `LoadingInstances` in each version will have only one key now: `typeof(TSelf)`. See experiment:  https://ideone.com/kXVsAi

But I think the `IsLoading` was a bit scary to use because it wasn't very apparent that it'd be defined "per non-generic implementation". e.g. https://stackoverflow.com/questions/9647641/resharper-warns-static-field-in-generic-type

Reply:
aahhh, that makes sense. ok, will add the other class back in.